packages/vite/src/node/server/ws.ts

```diff
@@ -28,9 +28,11 @@ export function createWebSocketServer(
   let httpsServer: Server | undefined = undefined
 
   const hmr = isObject(config.server.hmr) && config.server.hmr
-  const wsServer =
-    (hmr && hmr.server) ||
-    ((!(hmr && hmr.port) || hmr.port !== config.server.port) && server)
+  const hmrServer = hmr && hmr.server
+  const hmrPort = hmr && hmr.port
+  const portsAreCompatible =
+    !hmrPort || !config.server.port || hmrPort === config.server.port
```

Comment:
`config.server.port` is `undefined` for me, which means that `portsAreCompatible` is set to `true`. Since `hmrServer` is undefined, this results in `wsServer` being equal to `server`, and a new WS server isn't started.

Reply:
Good catch, I think it may be related to this https://github.com/vitejs/vite/pull/7318#issuecomment-1076223587. Check here https://github.com/vitejs/vite/blob/4e0b61593a0eba97a901f996619c5e1aae4a7b68/packages/vite/src/node/server/index.ts#L491.
We would need to await for this event, or this may require a bigger refactoring